pkg/wallet/box.go

```diff
@@ -0,0 +1,146 @@
+
+	"github.com/hyperledger/aries-framework-go/pkg/internal/cryptoutil"
+
+	"github.com/btcsuite/btcutil/base58"
+	"golang.org/x/crypto/nacl/box"
+)
+
+// Box provides an elliptic-curve-based authenticated encryption scheme
+//
+// Payloads are encrypted using symmetric encryption (XChacha20Poly1305)
+// using a shared key derived from a shared secret created by
+//   Curve25519 Elliptic Curve Diffie-Hellman key exchange.
+//
+// Box is created by a wallet, and reads secret keys from the wallet
+//   for encryption/decryption, so clients do not need to see
+//   the secrets themselves.
+type Box struct {
+	w *BaseWallet
+}
+
+// BoxEasy seals a message with a provided nonce
+// theirPub is used as a public key, while myPub is used to identify the private key that should be used
+func (b *Box) BoxEasy(payload, nonce []byte, theirPub, myPub string) ([]byte, error) {
+	recPubSlice := base58.Decode(theirPub)
+
+	var recPubBytes [cryptoutil.Curve25519KeySize]byte
+	copy(recPubBytes[:], recPubSlice)
+
+	//	 myPub is used to get the sender private key for encryption
+	kp, err := b.w.getKey(myPub)
+	if err != nil {
+		return nil, err
+	}
+
+	var priv [cryptoutil.Curve25519KeySize]byte
+	copy(priv[:], kp.Priv)
+
+	var nonceBytes [cryptoutil.NonceSize]byte
+	copy(nonceBytes[:], nonce)
+
+	ret := box.Seal(nil, payload, &nonceBytes, &recPubBytes, &priv)
+	return ret, nil
+}
+
+// BoxEasyOpen unseals a message sealed with BoxEasy, where the nonce is provided
+// theirPub is the public key used to decrypt directly, while myPub is used to identify the private key to be used
+func (b *Box) BoxEasyOpen(cipherText, nonce []byte, theirPub, myPub string) ([]byte, error) {
+	//	 myPub is used to get the recipient private key for decryption
+	sendPubSlice := base58.Decode(theirPub)
+
+	var sendPubBytes [cryptoutil.Curve25519KeySize]byte
+	copy(sendPubBytes[:], sendPubSlice)
+
+	kp, err := b.w.getKey(myPub)
+	if err != nil {
+		return nil, err
+	}
+
+	var priv [cryptoutil.Curve25519KeySize]byte
+	copy(priv[:], kp.Priv)
+
+	var nonceBytes [cryptoutil.NonceSize]byte
+	copy(nonceBytes[:], nonce)
+
+	out, success := box.Open(nil, cipherText, &nonceBytes, &sendPubBytes, &priv)
+	if !success {
+		return nil, errors.New("failed to unpack")
+	}
+
+	return out, nil
+}
+
+// BoxSeal seals a payload using the equivalent of libsodium box_seal
+//
+// Generates an ephemeral keypair to use for the sender, and includes
+// the ephemeral sender public key in the message.
+func (b *Box) BoxSeal(payload []byte, recPub string, randSource io.Reader) ([]byte, error) {
+	// generate ephemeral curve25519 asymmetric keys
+	epk, esk, err := box.GenerateKey(randSource)
+	if err != nil {
+		return nil, err
+	}
+
+	recPubSlice := base58.Decode(recPub)
+
+	var recPubBytes [cryptoutil.Curve25519KeySize]byte
+	copy(recPubBytes[:], recPubSlice)
+
+	nonce, err := cryptoutil.Nonce(epk[:], recPubSlice)
+	if err != nil {
+		return nil, err
+	}
+
+	var out = make([]byte, len(epk))
+	copy(out, epk[:])
```

Comment:
I think we can remove it 
```
var out = make([]byte, len(epk))
	copy(out, epk[:])
```
epk instead of out
and use `ret := box.Seal(epk, payload, nonce, &recPubBytes, esk)`